include/oneapi/dpl/pstl/unseq_backend_simd.h

```diff
@@ -823,7 +823,8 @@ __simd_find_first_of(_ForwardIterator1 __first, _ForwardIterator1 __last, _Forwa
     // Otherwise, vice versa.    
     if (__n1 < __n2)
     {
-        auto __u_pred = [__pred, __first](auto&& __val) { return __pred(__val, *__first); };
+        auto __u_pred =
+            [__pred, __first](auto&& __val) mutable { return __pred(std::forward<decltype(__val)>(__val), *__first); };
```

Comment:
Could you describe why `mutable` is needed? I do not understand why `__u_pred` should modify copies of `__pred` or `__first`.

Reply:
That lambda replaces `__equal_value_by_pred` structure.
`__u_pred `should not modify anything.
`mutable` because we have a test call with non-const predicate for that algo.

Reply:
I see, thanks. I confirm that the new function is an equivalent of `__equal_value_by_pred` in that regard.